database/gorm/query_test.go

```diff
@@ -2743,6 +2743,50 @@ func (s *QueryTestSuite) TestOrWhereIn() {
 	}
 }
 
+func (s *QueryTestSuite) TestWhereNotIn() {
+	for driver, query := range s.queries {
+		s.Run(driver.String(), func() {
+			user := User{Name: "where_in_user", Avatar: "where_in_avatar"}
+			s.Nil(query.Create(&user))
+			s.True(user.ID > 0)
+
+			user1 := User{Name: "where_in_user_1", Avatar: "where_in_avatar_1"}
+			s.Nil(query.Create(&user1))
+			s.True(user1.ID > 0)
+
+			user2 := User{Name: "where_in_user_2", Avatar: "where_in_avatar_2"}
+			s.Nil(query.Create(&user2))
+			s.True(user2.ID > 0)
+
+			var users []User
+			s.Nil(query.WhereNotIn("id", []any{user.ID, user1.ID}).Find(&users))
```

Comment:
S: We'd better add a `Where` condition with `WhereNotIn`, only get the `user2` row, it will be more accurate. So the `Avatar` of these three SQL can be the same.

Reply:
Done.